server/server/src/main/java/org/infinispan/server/server/Bootstrap.java

```diff
@@ -0,0 +1,124 @@
+      File configurationFile = new File(confDir, Server.DEFAULT_CONFIGURATION_FILE);
+      File logDir = new File(serverRoot, Server.DEFAULT_SERVER_LOG);
+      properties.putIfAbsent(Server.INFINISPAN_SERVER_LOG, logDir.getAbsolutePath());
+      try (InputStream is = new FileInputStream(new File(confDir, "logging.properties"))) {
+         LogManager.getLogManager().readConfiguration(is);
+      } catch (IOException e) {
+         stdErr.printf("Could not load logging.properties: %s", e.getMessage());
+         e.printStackTrace(stdErr);
+      }
+
+      for (int i = 0; i < args.length; i++) {
+         String command = args[i];
+         String parameter = null;
+         if (command.startsWith("--")) {
+            int equals = command.indexOf('=');
+            if (equals > 0) {
+               parameter = command.substring(equals + 1);
+               command = command.substring(0, equals);
+            }
+         } else if (command.startsWith("-")) {
+            if (command.length() != 2) {
+               stdErr.println(MSG.invalidShortArgument(command));
+               exitHandler.exit(1);
+            }
+         } else {
+            stdErr.println(MSG.invalidArgument(command));
+            exitHandler.exit(1);
+         }
+         switch (command) {
+            case "-h":
+            case "--help":
+               help(stdOut, true);
+               exitHandler.exit(0);
+               break;
+            case "-c":
+               parameter = args[++i];
+               // Fall through
+            case "--server-config":
+               configurationFile = new File(parameter);
+               break;
+            case "--s":
+               parameter = args[++i];
+               // Fall through
+            case "--server-root":
+               serverRoot = new File(parameter);
+               break;
+            case "-b":
+               parameter = args[++i];
+               // Fall through
+            case "--bind-address":
+               properties.setProperty(Server.INFINISPAN_BIND_ADDRESS, parameter);
+               break;
+            case "-o":
+               parameter = args[++i];
+               // Fall through
+            case "--port-offset":
+               properties.setProperty(Server.INFINISPAN_PORT_OFFSET, parameter);
+               break;
+            case "-v":
+            case "--version":
+               help(stdOut, false);
+               exitHandler.exit(0);
+               break;
+            default:
+               stdErr.println(MSG.unknownParameter(args[i]));
+               exitHandler.exit(1);
+               break;
+         }
+      }
+      try {
+         Runtime.getRuntime().addShutdownHook(new ShutdownHook(exitHandler));
+         Server.log.serverStarting(Version.getBrandName());
+         Server server = new Server(serverRoot, configurationFile, properties);
+         server.setExitHandler(exitHandler);
+         server.run().get();
```

Comment:
We surely spawn some user threads when starting Infinispan? I don't see why we have to block the main thread from completing.

Reply:
Hmmm, I don't know what's the best thing to do here, but I'd just check the what WildFly does here.